### PR TITLE
Update dependencies, notably Go 1.18

### PR DESCRIPTION
**Pull request by codefromthecrypt**
See https://go.dev/blog/go1.18

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
looks like there are two problems.

1. go removed the ability to `go run` a version
```
package github.com/golangci/golangci-lint/cmd/golangci-lint@v1.44.2: can only use path@version syntax with 'go get'
[8](https://github.com/tetratelabs/car/runs/5578537766?check_suite_focus=true#step:4:8)
```

2. linter doesn't yet support go 1.18 https://github.com/golangci/golangci-lint/issues/2649

**Comment by codefromthecrypt** [comment]
actually go 1.18 isn't on the runner, yet https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#go either

**Comment by codefromthecrypt** [comment]
@mathetake PTAL, figured I'd update car and func-e to go 1.18, now that things work.

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,15 @@
 module github.com/tetratelabs/car
 
 // This project uses go:embed, so requires minimally go 1.16
-go 1.17
+go 1.18
```
Comment (codefromthecrypt) [inline go.mod:4]: FYI: this is read by golangci-lint to know whether or not to disable <1.18 linters

**Review by mathetake — APPROVED** [review]

**Pull request merged**